x-pack/plugins/security_solution/public/common/components/toolbar/bulk_actions/translations.ts

```diff
@@ -7,18 +7,26 @@
 
 import { i18n } from '@kbn/i18n';
 
-export const SELECTED_ALERTS = (selectedAlertsFormatted: string, selectedAlerts: number) =>
-  i18n.translate('xpack.securitySolution.toolbar.bulkActions.selectedAlertsTitle', {
-    values: { selectedAlertsFormatted, selectedAlerts },
+export const SELECTED_ENTITIES = (
+  entity: string,
+  selectedAlertsFormatted: string,
+  selectedAlerts: number
+) =>
+  i18n.translate('xpack.securitySolution.toolbar.bulkActions.selectedEntitiesTitle', {
+    values: { entity, selectedAlertsFormatted, selectedAlerts },
     defaultMessage:
-      'Selected {selectedAlertsFormatted} {selectedAlerts, plural, =1 {alert} other {alerts}}',
+      'Selected {selectedAlertsFormatted} {entity}{selectedAlerts, plural, =1 {} other {s}}',
   });
 
-export const SELECT_ALL_ALERTS = (totalAlertsFormatted: string, totalAlerts: number) =>
-  i18n.translate('xpack.securitySolution.toolbar.bulkActions.selectAllAlertsTitle', {
-    values: { totalAlertsFormatted, totalAlerts },
+export const SELECT_ALL_ENTITIES = (
+  entity: string,
+  totalAlertsFormatted: string,
+  totalAlerts: number
+) =>
+  i18n.translate('xpack.securitySolution.toolbar.bulkActions.selectAllEntitiesTitle', {
+    values: { entity, totalAlertsFormatted, totalAlerts },
     defaultMessage:
-      'Select all {totalAlertsFormatted} {totalAlerts, plural, =1 {alert} other {alerts}}',
+      'Select {totalAlerts, plural, =1 {{totalAlertsFormatted} {entity}} other {all {totalAlertsFormatted} {entity}s} }',
```

Comment:
In the previous version, the `alert`|`alerts` text belonged to the translation string and could be translated, now the `entity` string is a constant received by params, that can not be translated.

There are different ways to make the entities' text translatable, we could create separate eventType translations locally, something like that:

```
const ENTITY_TYPE_PLURAL = (entityType: TableEntityType, count: number) => {
  switch (entityType) {
    case TableEntityType.alert: {
      return i18n.translate('xpack.securitySolution.toolbar.bulkActions.entityAlerts', {
        values: { count },
        defaultMessage: '{count, plural, =1 {alert} other {alerts}}',
      });
    }
    case TableEntityType.event: {
      return i18n.translate('xpack.securitySolution.toolbar.bulkActions.entityEvents', {
        values: { count },
        defaultMessage: '{count, plural, =1 {event} other {events}}',
      });
    }
    case TableEntityType.session: {
      return i18n.translate('xpack.securitySolution.toolbar.bulkActions.entitySessions', {
        values: { count },
        defaultMessage: '{count, plural, =1 {session} other {sessions}}',
      });
    }
  }
};
```
Then in these translations we can do:
```
export const SELECT_ALL_ENTITIES = (
  entityType: TableEntityType,
  totalFormatted: string,
  total: number
) =>
  i18n.translate('xpack.securitySolution.toolbar.bulkActions.selectAllEntitiesTitle', {
    values: { entityPlural: ENTITY_TYPE_PLURAL(entityType, total), totalFormatted },
    defaultMessage:
      'Select {totalFormatted} {entityPlural}',
```

Or using an object of translation functions having only `count` param, and indexed by `TableEntityType`. That would also work.

Reply:
Thanks Sergi! That makes sense, I hadn't considered the entityType would be constant, good eye. I made that change. :D